cpp/daal/src/data_management/service_numeric_table.h

```diff
@@ -396,6 +398,75 @@ class SOANumericTableCPU : public SOANumericTable
         DAAL_DEFAULT_CREATE_TEMPLATE_IMPL_EX(SOANumericTableCPU, DAAL_TEMPLATE_ARGUMENTS(cpu), ddict, nRows, memoryAllocationFlag);
     }
 
+    template <typename T>
+    services::Status getTFeature(size_t feat_idx, size_t idx, size_t nrows, int rwFlag, BlockDescriptor<T> & block)
+    {
+        size_t nobs = getNumberOfRows();
+        block.setDetails(feat_idx, idx, rwFlag);
+
+        if (idx >= nobs)
+        {
+            block.resizeBuffer(1, 0);
+            return services::Status();
+        }
+
+        nrows = (idx + nrows < nobs) ? nrows : nobs - idx;
+
+        const NumericTableFeature & f = (*_ddict)[feat_idx];
+        const int indexType           = f.indexType;
+
+        if (features::internal::getIndexNumType<T>() == f.indexType)
+        {
+            block.setPtr(&(_arrays[feat_idx]), _arrays[feat_idx].get() + idx * f.typeSize, 1, nrows);
+        }
+        else
+        {
+            if (data_management::features::DAAL_OTHER_T == indexType)
+            {
+                block.reset();
+                return services::Status(services::ErrorDataTypeNotSupported);
+            }
+
+            byte * location = _arrays[feat_idx].get() + idx * f.typeSize;
+            if (!block.resizeBuffer(1, nrows))
+            {
+                return services::Status(services::ErrorMemoryAllocationFailed);
+            }
+
+            if (!(block.getRWFlag() & (int)readOnly)) return services::Status();
+
+            getVectorUpCast(indexType, getConversionDataType<T>())(nrows, location, block.getBlockPtr());
+        }
+        return services::Status();
+    }
+
+    template <typename T>
+    services::Status releaseTFeature(BlockDescriptor<T> & block)
+    {
+        if (block.getRWFlag() & (int)writeOnly)
```

Comment:
Maybe it can be more reliable to use direct comparison? Smth like `if((block.RWFlag() == writeOnly) || (block.RWFlag == readWrite))`